### DISCUSSION TITLE
Thoughts about Odometer plugin

### DISCUSSION THREAD

**Discussion by jameslkey** [grblHAL/core#21] [Ideas]
I'm currently looking at creating the interface for the odometer in IOSender. One thought I had was should the plug-in be expanded to enable the possibility of the 3² (9 for the math inhibited) possible axes? It would make some differences to the gui as well as possible customizations at compile time.

**Comment by terjeio** [comment]
The plugin handles whatever number of axes the core is configured for, up to 6 currently. ioSender is limited to 6 too, both the grblHAL core and the ioSender core + relevant controls needs to be changed to support more than 6. grblHAL driver(s) and controller hardware to match has to be updated/created as well.

What are the use cases for increasing the number of axes supported to more than 6? And is it worth putting it on the todo list?

**Comment by jameslkey** [comment]
Good question, I was looking at the possibilities listed on the driver
section and noticed the "3²". I can only imagine a device that would have
the bass x,y,z plus angular rotation of the head, pivoting, and plunging.
Then 'table" rotation, in xy, and a rotating chuck like a lathe. So 8, but
I can't see too many people needing that. So probably no good reason except
is it easier to implement now or later and how much code would needed to be
maintained? On the plugin side I see very little. But on the other side
maybe too much.

I did notice a possible issue in the plugin code. It seems that currently
the C_AXIS only gets updated if there is a change in the B axis. One letter
typo?

**Reply by terjeio** [reply]
> Good question, I was looking at the possibilities listed on the driver section and noticed the "3²".

Oh I understand now: the 2 superscript refers to footnote 2...

> So probably no good reason except
is it easier to implement now or later and how much code would needed to be
maintained? On the plugin side I see very little. But on the other side
maybe too much.

It will be a rather large piece of work, I have a lot of other stuff to complete before considering increasing the number of axes supported... 

> I did notice a possible issue in the plugin code. It seems that currently
the C_AXIS only gets updated if there is a change in the B axis. One letter
typo?

Thanks for spotting that, yes it is a typo - I'll commit a fix later.

**Comment by jameslkey** [comment]
Lol, at least I got that right. I see the ² now! Okay then off I go to see
if I can get started!
